### PR TITLE
Amanda's Game of Life

### PR DESCRIPTION
**Pull request by amandena**
Video Demo: https://www.dropbox.com/s/ppv65n3rxbyytl8/GameOfLife-Walkthrough.mov?dl=0

### CODE REVIEW

File: amandas-project/src/components/Cell.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+const CELL_SIZE = 20
```
Comment (carloshdelreal) [inline amandas-project/src/components/Cell.js:3]: It is a better practice to define all of this variables on a separate file or in the main file to be able to easily change them if required

File: amandas-project/src/containers/Game.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import React from 'react'
+import Cell from '../components/Cell'
+
+const CELL_SIZE = 20
+const WIDTH = 400
+const HEIGHT = 400
```
Comment (carloshdelreal) [inline amandas-project/src/containers/Game.js:6]: like this three guys over here

File: amandas-project/src/containers/Game.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+class Game extends React.Component {
+  constructor() {
+    super()
+    this.rows = HEIGHT / CELL_SIZE
+    this.cols = WIDTH / CELL_SIZE
+    this.board = this.makeEmptyBoard()
+  }
+
+  state = {
```
Comment (carloshdelreal) [inline amandas-project/src/containers/Game.js:16]: the state has to be inside of the constructor and be used like this.state = {}

Comment (carloshdelreal) [reply amandas-project/src/containers/Game.js:16]: does this work??

File: amandas-project/src/containers/Game.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+          } else {
+            newBoard[y][x] = false
+          }
+        } else {
+          if(!this.board[y][x] && neighbors === 3) {
+            newBoard[y][x] = true
+          }
+        }
+      }
+    }
```
Comment (carloshdelreal) [inline amandas-project/src/containers/Game.js:57]: I really would like you to have all of this game logic code on a separate file and not here inside the game component, this makes your code very difficult to understand

File: amandas-project/src/containers/Game.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    for(let i = 0; i < dirs.length; i++) {
+      const dir = dirs[i]
+      let y1 = y + dir[0]
+      let x1 = x + dir[1]
+      if(x1 >= 0 && x1 < this.cols && y1 >= 0 && y1 < this.rows && board[y1][x1]) {
+        neighbors++
+      }
+    }
+    return neighbors
+  }
```
Comment (carloshdelreal) [inline amandas-project/src/containers/Game.js:80]: this function in other file too, this is game logic

**Review by carloshdelreal — COMMENTED** [review]
I really like you created a cell component and a game component but it would be nice if you try to split the rendering part of the application (the components) and the game logic, your game.js component could be reduced to something more readable. 
besides that I would like you to add some tests!!, are you sure any corner case breaks your code??

File: amandas-project/src/containers/Game.js [hunk 6]
```diff
@@ -0,0 +1,160 @@
+import React from 'react'
+import Cell from '../components/Cell'
+
+const CELL_SIZE = 20
+const WIDTH = 400
+const HEIGHT = 400
+
+class Game extends React.Component {
```
Comment (agarci1994) [inline amandas-project/src/containers/Game.js:8]: I would recommend using functional components with hooks, rather than class components. They are more modern, cleaner and are currently more widely used.

**Review by agarci1994 — COMMENTED** [review]
I see quite clean and well organized, it seems to work well, I can't find any more bugs
